src/debug/inc/arm/primitives.h

```diff
@@ -30,7 +30,11 @@ typedef DPTR(CORDB_ADDRESS_TYPE)    PTR_CORDB_ADDRESS_TYPE;
 #define STACKWALK_CONTROLPC_ADJUST_OFFSET 2
 
 #define CORDbg_BREAK_INSTRUCTION_SIZE 2
+#ifdef LINUX32
```

Comment:
Actually, I am not sure because I do not have any non-Linux ARM devices. But since we observe different behavior for Linux on x86 vs ARM, I guess this is platform/OS specific. So I think `PC` modification should be under `#ifdef __linux__` because I can confirm such behavior only for ARM Linux.